apps/ejabberd/src/ejabberd_local_delivery.erl

```diff
@@ -0,0 +1,34 @@
+%%%-------------------------------------------------------------------
+%%% @author bartek
+%%% @copyright (C) 2016, <COMPANY>
+%%% @doc
+%%%
+%%% @end
+%%% Created : 21. Mar 2016 12:24
+%%%-------------------------------------------------------------------
+-module(ejabberd_local_delivery).
+-author("bartek").
+
+-include("jlib.hrl").
+
+%% API
+-export([do_local_route/5]).
+
+
+do_local_route(OrigFrom, OrigTo, OrigPacket, LDstDomain, Handler) ->
```

Comment:
I suggest changing the function name to just `do_route`. I think this invocation will be informative enough:
```erlang
ejabberd_local_deliver:do_route(From, To, Packet, Domain, Handler)
```